index.js

```diff
@@ -80,23 +82,46 @@ fetch(apiUrl)
         iconArray.push(data.list[2].weather[0].icon);
         iconArray.push(data.list[10].weather[0].icon);
         iconArray.push(data.list[18].weather[0].icon);
-    })
 
-/* tried to fetch api url from flicker for changing background images,but encountering with an error */
+        function createIcon() {
+            let iconSet = document.querySelectorAll("ul > li");
+            for (let i = 0; i < iconArray.length && i < iconSet.length; i++) {
+                let newIconEle = document.createElement("span");
+                newIconEle.classList.add("iconStyle");
+                let imageUrl = "http://openweathermap.org/img/wn/" + iconArray[i] + "@2x.png";
+                newIconEle.innerHTML = "<img src=" + imageUrl + ">";
+                iconSet[i].appendChild(newIconEle);
+            }
+        }
+        createIcon();
+    });
 searchBtnCmp.addEventListener("click", function() {
-
-    const cityUrl = "http://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=528fc5e2ec09109cd0696b0555e7bc81&tags=" + weatherSearchBar.value + "&format=json&nojsoncallback=1";
+    const proxyurl = "https://cors-anywhere.herokuapp.com/";
+    const cityUrl = "http://api.flickr.com/services/rest/?method=flickr.photos.search&api_key=ef3d8a6207b554e853fcfe4b900bd9b7&tags=" + weatherSearchBar.value + "&format=json&nojsoncallback=1";
     console.log(cityUrl);
-    fetch(cityUrl)
+    fetch(proxyurl + cityUrl)
         .then(response => response.json())
         .then(data => {
 
-            let farmId = photos.photo[0].id;
-            let serverId = photos.photo[0].server;
-            let id = photos.photo[0].id;
-            let secretId = photos.photo[0].secret;
+            let farmId = data.photos.photo[0].id;
+            let serverId = data.photos.photo[0].server;
+            let id = data.photos.photo[0].id;
+            let secretId = data.photos.photo[0].secret;
             let backgroundImageUrl = "https://farm" + farmId + ".staticflickr.com/" + serverId + "/" + id + "_" + secretId + ".jpg";
```

Comment:
you can do it better like this:
```js
  const backgroundImageUrl = `https://farm${farmId}.staticflickr.com/${serverId}/${id}_${secretId}.jpg`;
```
and in general - if you don't plan to change variable => make it a constant;

Could you please point out to docs where I can read more about this url construction pattern. Because server can't be reached in all cases.

Reply:
Ah ok, I found a typo. 
In this line `let farmId = data.photos.photo[0].id;` it should be not `id` but 'farm'. But looks like you can't rely on this api, because you would get not really what you want for some reason...Search for London didn't return any images about london city. but some people...
I'll try another api myself and will update in here.

Reply:
Ok, seems like it's working with flickr so let's keep it as it is. Please adjust this and comment about how you set an background image and then you're good to go with the images.

Reply:
But still,after correcting all these,im still getting no access to the image link "farm49999694212.staticflickr.com’s server IP address could not be found." This is what im getting

Reply:
But i just managed to get the background image using another api called pixabay,but i just saw the image url is valid only for 24 hrs but its working